### PR TITLE
Add test to verify getServiceList behavior

### PR DESCRIPTION
**Pull request by LucasRoesler**
## Description

**What**
- Add unit test that verifies that getServiceList correctly finds
  and filters deployments to just function objects.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Relates to #738 

## How Has This Been Tested?

It is a test

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by LucasRoesler** [comment]
The failing test seems to _only_ fail during the arm docker builds.  Master builds just fine and it is only my test that it is failing. It behaves like the k8s mock client is never initialized at all

```sh
#33 70.19     reader_test.go:104: expected 2 functions, got 0
```

When I run the tests on my laptop, they pass.

```sh
?   	github.com/openfaas/faas-netes	[no test files]
?   	github.com/openfaas/faas-netes/pkg/apis/openfaas	[no test files]
?   	github.com/openfaas/faas-netes/pkg/apis/openfaas/v1	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/fake	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/scheme	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/typed/openfaas/v1	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/typed/openfaas/v1/fake	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/internalinterfaces	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/openfaas	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/openfaas/v1	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/listers/openfaas/v1	[no test files]
ok  	github.com/openfaas/faas-netes/pkg/config	0.017s
ok  	github.com/openfaas/faas-netes/pkg/controller	0.041s
ok  	github.com/openfaas/faas-netes/pkg/handlers	0.057s
ok  	github.com/openfaas/faas-netes/pkg/k8s	0.036s
ok  	github.com/openfaas/faas-netes/pkg/server	0.042s
?   	github.com/openfaas/faas-netes/pkg/signals	[no test files]
?   	github.com/openfaas/faas-netes/version	[no test files]
```

When I run it on my PI, they pass.

```sh
?   	github.com/openfaas/faas-netes	[no test files]
?   	github.com/openfaas/faas-netes/pkg/apis/openfaas	[no test files]
?   	github.com/openfaas/faas-netes/pkg/apis/openfaas/v1	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/fake	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/scheme	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/typed/openfaas/v1	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/typed/openfaas/v1/fake	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/internalinterfaces	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/openfaas	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/openfaas/v1	[no test files]
?   	github.com/openfaas/faas-netes/pkg/client/listers/openfaas/v1	[no test files]
ok  	github.com/openfaas/faas-netes/pkg/config	0.017s
ok  	github.com/openfaas/faas-netes/pkg/controller	0.041s
ok  	github.com/openfaas/faas-netes/pkg/handlers	0.057s
ok  	github.com/openfaas/faas-netes/pkg/k8s	0.036s
ok  	github.com/openfaas/faas-netes/pkg/server	0.042s
?   	github.com/openfaas/faas-netes/pkg/signals	[no test files]
?   	github.com/openfaas/faas-netes/version	[no test files]

```

Oddly, I can only see it fail during `make build-buildx-all` 
```sh
 => ERROR [linux/amd64 build 7/8] RUN CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go test -v ./...                    70.5s
------
 > [linux/amd64 build 7/8] RUN CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go test -v ./...:
#33 67.17 ?   	github.com/openfaas/faas-netes	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/apis/openfaas	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/apis/openfaas/v1	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/fake	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/scheme	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/typed/openfaas/v1	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/clientset/versioned/typed/openfaas/v1/fake	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/internalinterfaces	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/openfaas	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/informers/externalversions/openfaas/v1	[no test files]
#33 67.17 ?   	github.com/openfaas/faas-netes/pkg/client/listers/openfaas/v1	[no test files]
#33 67.17 === RUN   TestRead_EmptyProbeConfig
#33 67.17 --- PASS: TestRead_EmptyProbeConfig (0.00s)
#33 67.17 === RUN   TestRead_HTTPProbeConfig
#33 67.17 --- PASS: TestRead_HTTPProbeConfig (0.00s)
#33 67.17 === RUN   TestRead_HTTPProbeConfig_true
#33 67.17 --- PASS: TestRead_HTTPProbeConfig_true (0.00s)
#33 67.17 === RUN   TestRead_ImagePullPolicy_set
#33 67.17 --- PASS: TestRead_ImagePullPolicy_set (0.00s)
#33 67.17 === RUN   TestRead_ImagePullPolicy_empty
#33 67.17 --- PASS: TestRead_ImagePullPolicy_empty (0.00s)
#33 67.17 PASS
#33 67.17 ok  	github.com/openfaas/faas-netes/pkg/config	0.429s
#33 70.19 === RUN   Test_makeAnnotations_NoKeys
#33 70.19 --- PASS: Test_makeAnnotations_NoKeys (0.01s)
#33 70.19 === RUN   Test_makeAnnotations_WithKeyAndValue
#33 70.19 --- PASS: Test_makeAnnotations_WithKeyAndValue (0.00s)
#33 70.19 === RUN   Test_makeAnnotationsDoesNotModifyOriginalSpec
#33 70.19 --- PASS: Test_makeAnnotationsDoesNotModifyOriginalSpec (0.00s)
#33 70.19 === RUN   Test_newDeployment
#33 70.19 I0115 21:46:04.690199    8766 deployment.go:147] Function kubesec: no profiles specified
#33 70.19 I0115 21:46:04.696897    8766 deployment.go:157] Function kubesec: Applying profiles []
#33 70.19 --- PASS: Test_newDeployment (0.02s)
#33 70.19 === RUN   Test_newDeployment_PrometheusScrape_NotOverridden
#33 70.19 I0115 21:46:04.702360    8766 deployment.go:147] Function kubesec: no profiles specified
#33 70.19 I0115 21:46:04.702458    8766 deployment.go:157] Function kubesec: Applying profiles []
#33 70.19 --- PASS: Test_newDeployment_PrometheusScrape_NotOverridden (0.00s)
#33 70.19 === RUN   Test_Replicas
#33 70.19 === RUN   Test_Replicas/return_nil_replicas_when_label_is_missing_and_deployment_does_not_exist
#33 70.19 I0115 21:46:04.705174    8766 deployment.go:147] Function : no profiles specified
#33 70.19 I0115 21:46:04.705232    8766 deployment.go:157] Function : Applying profiles []
#33 70.19 === RUN   Test_Replicas/return_nil_replicas_when_label_is_missing_and_deployment_has_no_replicas
#33 70.19 I0115 21:46:04.706048    8766 deployment.go:147] Function : no profiles specified
#33 70.19 I0115 21:46:04.706109    8766 deployment.go:157] Function : Applying profiles []
#33 70.19 === RUN   Test_Replicas/return_min_replicas_when_label_is_present_and_deployment_has_nil_replicas
#33 70.19 I0115 21:46:04.706912    8766 deployment.go:147] Function : no profiles specified
#33 70.19 I0115 21:46:04.706990    8766 deployment.go:157] Function : Applying profiles []
#33 70.19 === RUN   Test_Replicas/return_min_replicas_when_label_is_present_and_deployment_has_replicas_less_than_min
#33 70.19 I0115 21:46:04.707502    8766 deployment.go:147] Function : no profiles specified
#33 70.19 I0115 21:46:04.707560    8766 deployment.go:157] Function : Applying profiles []
#33 70.19 === RUN   Test_Replicas/return_existing_replicas_when_label_is_present_and_deployment_has_more_replicas_than_min
#33 70.19 I0115 21:46:04.707961    8766 deployment.go:147] Function : no profiles specified
#33 70.19 I0115 21:46:04.708007    8766 deployment.go:157] Function : Applying profiles []
#33 70.19 === RUN   Test_Replicas/return_existing_replicas_when_label_is_missing_and_deployment_has_replicas_set_by_HPA
#33 70.19 I0115 21:46:04.708310    8766 deployment.go:147] Function : no profiles specified
#33 70.19 I0115 21:46:04.708353    8766 deployment.go:157] Function : Applying profiles []
#33 70.19 === RUN   Test_Replicas/return_zero_replicas_when_label_is_present_and_deployment_has_zero_replicas
#33 70.19 I0115 21:46:04.708724    8766 deployment.go:147] Function : no profiles specified
#33 70.19 I0115 21:46:04.708772    8766 deployment.go:157] Function : Applying profiles []
#33 70.19 --- PASS: Test_Replicas (0.01s)
#33 70.19     --- PASS: Test_Replicas/return_nil_replicas_when_label_is_missing_and_deployment_does_not_exist (0.00s)
#33 70.19     --- PASS: Test_Replicas/return_nil_replicas_when_label_is_missing_and_deployment_has_no_replicas (0.00s)
#33 70.19     --- PASS: Test_Replicas/return_min_replicas_when_label_is_present_and_deployment_has_nil_replicas (0.00s)
#33 70.19     --- PASS: Test_Replicas/return_min_replicas_when_label_is_present_and_deployment_has_replicas_less_than_min (0.00s)
#33 70.19     --- PASS: Test_Replicas/return_existing_replicas_when_label_is_present_and_deployment_has_more_replicas_than_min (0.00s)
#33 70.19     --- PASS: Test_Replicas/return_existing_replicas_when_label_is_missing_and_deployment_has_replicas_set_by_HPA (0.00s)
#33 70.19     --- PASS: Test_Replicas/return_zero_replicas_when_label_is_present_and_deployment_has_zero_replicas (0.00s)
#33 70.19 === RUN   Test_UpdateSecrets_DoesNotAddVolumeIfRequestSecretsIsNil
#33 70.19 --- PASS: Test_UpdateSecrets_DoesNotAddVolumeIfRequestSecretsIsNil (0.00s)
#33 70.19 === RUN   Test_UpdateSecrets_DoesNotAddVolumeIfRequestSecretsIsEmpty
#33 70.19 --- PASS: Test_UpdateSecrets_DoesNotAddVolumeIfRequestSecretsIsEmpty (0.00s)
#33 70.19 === RUN   Test_UpdateSecrets_RemovesAllCopiesOfExitingSecretsVolumes
#33 70.19 --- PASS: Test_UpdateSecrets_RemovesAllCopiesOfExitingSecretsVolumes (0.00s)
#33 70.19 === RUN   Test_UpdateSecrets_AddNewSecretVolume
#33 70.19 --- PASS: Test_UpdateSecrets_AddNewSecretVolume (0.00s)
#33 70.19 === RUN   Test_UpdateSecrets_ReplacesPreviousSecretMountWithNewMount
#33 70.19 --- PASS: Test_UpdateSecrets_ReplacesPreviousSecretMountWithNewMount (0.00s)
#33 70.19 === RUN   Test_UpdateSecrets_RemovesSecretsVolumeIfRequestSecretsIsEmptyOrNil
#33 70.19 --- PASS: Test_UpdateSecrets_RemovesSecretsVolumeIfRequestSecretsIsEmptyOrNil (0.00s)
#33 70.19 PASS
#33 70.19 ok  	github.com/openfaas/faas-netes/pkg/controller	0.490s
#33 70.19 === RUN   Test_buildAnnotations_Empty_In_CreateRequest
#33 70.19 --- PASS: Test_buildAnnotations_Empty_In_CreateRequest (0.00s)
#33 70.19 === RUN   Test_buildAnnotations_Premetheus_NotOverridden
#33 70.19 --- PASS: Test_buildAnnotations_Premetheus_NotOverridden (0.00s)
#33 70.19 === RUN   Test_buildAnnotations_From_CreateRequest
#33 70.19 --- PASS: Test_buildAnnotations_From_CreateRequest (0.00s)
#33 70.19 === RUN   Test_SetNonRootUser
#33 70.19 === RUN   Test_SetNonRootUser/does_not_set_userid_value_when_SetNonRootUser_is_false
#33 70.19 === RUN   Test_SetNonRootUser/does_set_userid_to_constant_value_when_SetNonRootUser_is_true
#33 70.19 --- PASS: Test_SetNonRootUser (0.01s)
#33 70.19     --- PASS: Test_SetNonRootUser/does_not_set_userid_value_when_SetNonRootUser_is_false (0.01s)
#33 70.19     --- PASS: Test_SetNonRootUser/does_set_userid_to_constant_value_when_SetNonRootUser_is_true (0.00s)
#33 70.19 === RUN   Test_buildEnvVars_NoSortedKeys
#33 70.19 --- PASS: Test_buildEnvVars_NoSortedKeys (0.00s)
#33 70.19 === RUN   Test_buildEnvVars_TwoSortedKeys
#33 70.19 --- PASS: Test_buildEnvVars_TwoSortedKeys (0.00s)
#33 70.19 === RUN   Test_buildEnvVars_FourSortedKeys
#33 70.19 --- PASS: Test_buildEnvVars_FourSortedKeys (0.00s)
#33 70.19 === RUN   Test_DefaultErrorReason
#33 70.19 --- PASS: Test_DefaultErrorReason (0.00s)
#33 70.19 === RUN   Test_ConflictErrorReason
#33 70.19 --- PASS: Test_ConflictErrorReason (0.00s)
#33 70.19 === RUN   Test_ValidateDeployRequest_ValidCharacters
#33 70.19 --- PASS: Test_ValidateDeployRequest_ValidCharacters (0.00s)
#33 70.19 === RUN   Test_ValidateDeployRequest_InvalidCharacters
#33 70.19 --- PASS: Test_ValidateDeployRequest_InvalidCharacters (0.00s)
#33 70.19 === RUN   Test_InfoHandler
#33 70.19 --- PASS: Test_InfoHandler (0.01s)
#33 70.19 === RUN   Test_findNamespace_Found
#33 70.19 --- PASS: Test_findNamespace_Found (0.00s)
#33 70.19 === RUN   Test_findNamespace_NotFound
#33 70.19 --- PASS: Test_findNamespace_NotFound (0.00s)
#33 70.19 === RUN   Test_getServiceList_ReturnsOnlyLabeledFunctions
#33 70.19     reader_test.go:104: expected 2 functions, got 0
#33 70.19 --- FAIL: Test_getServiceList_ReturnsOnlyLabeledFunctions (0.04s)
#33 70.19 === RUN   Test_SecretsHandler
#33 70.19 === RUN   Test_SecretsHandler/create_managed_secrets
#33 70.19 2021/01/15 21:46:04 created secret testsecret.of-fnc
#33 70.19 2021/01/15 21:46:04 Secret testsecret create
#33 70.19 2021/01/15 21:46:04 failed to create secret testsecret.of-fnc: secrets "testsecret" already exists
#33 70.19 2021/01/15 21:46:04 Secret create error reason: AlreadyExists, secrets "testsecret" already exists
#33 70.19 === RUN   Test_SecretsHandler/update_managed_secrets
#33 70.19 2021/01/15 21:46:04 Secret testsecret updated
#33 70.19 === RUN   Test_SecretsHandler/update_non-existing_secrets_returns_404
#33 70.19 2021/01/15 21:46:04 can not retrieve secret for update testsecret-doesnotexist.of-fnc: secrets "testsecret-doesnotexist" not found
#33 70.19 2021/01/15 21:46:04 Secret update error reason: NotFound, secrets "testsecret-doesnotexist" not found
#33 70.19 === RUN   Test_SecretsHandler/list_managed_secrets_only
#33 70.19 === RUN   Test_SecretsHandler/delete_managed_secrets
#33 70.19 2021/01/15 21:46:04 Secret testsecret deleted
#33 70.19 === RUN   Test_SecretsHandler/delete_missing_secret_returns_404
#33 70.19 2021/01/15 21:46:04 can not delete testsecret-doesnotexist.of-fnc: secrets "testsecret-doesnotexist" not found
#33 70.19 2021/01/15 21:46:04 Secret delete error reason: NotFound, secrets "testsecret-doesnotexist" not found
#33 70.19 --- PASS: Test_SecretsHandler (0.05s)
#33 70.19     --- PASS: Test_SecretsHandler/create_managed_secrets (0.03s)
#33 70.19     --- PASS: Test_SecretsHandler/update_managed_secrets (0.00s)
#33 70.19     --- PASS: Test_SecretsHandler/update_non-existing_secrets_returns_404 (0.00s)
#33 70.19     --- PASS: Test_SecretsHandler/list_managed_secrets_only (0.01s)
#33 70.19     --- PASS: Test_SecretsHandler/delete_managed_secrets (0.00s)
#33 70.19     --- PASS: Test_SecretsHandler/delete_missing_secret_returns_404 (0.00s)
#33 70.19 === RUN   Test_SecretsHandler_ListEmpty
#33 70.19 --- PASS: Test_SecretsHandler_ListEmpty (0.00s)
#33 70.19 === RUN   Test_NamespaceResolver
#33 70.19 === RUN   Test_NamespaceResolver/no_custom_namespace
#33 70.19 === RUN   Test_NamespaceResolver/allowed_namespace_in_GET_query_parameters
#33 70.19 === RUN   Test_NamespaceResolver/forbidden_namespace_in_GET_query_parameters
#33 70.19 === RUN   Test_NamespaceResolver/allowed_namespace_in_POST_body
#33 70.19 === RUN   Test_NamespaceResolver/forbidden_namespace_in_POST_body
#33 70.19 --- PASS: Test_NamespaceResolver (0.01s)
#33 70.19     --- PASS: Test_NamespaceResolver/no_custom_namespace (0.00s)
#33 70.19     --- PASS: Test_NamespaceResolver/allowed_namespace_in_GET_query_parameters (0.00s)
#33 70.19     --- PASS: Test_NamespaceResolver/forbidden_namespace_in_GET_query_parameters (0.00s)
#33 70.19     --- PASS: Test_NamespaceResolver/allowed_namespace_in_POST_body (0.00s)
#33 70.19     --- PASS: Test_NamespaceResolver/forbidden_namespace_in_POST_body (0.00s)
#33 70.19 FAIL
#33 70.19 FAIL	github.com/openfaas/faas-netes/pkg/handlers	0.629s
#33 70.20 === RUN   Test_AsFunctionStatus
#33 70.20 --- PASS: Test_AsFunctionStatus (0.00s)
#33 70.20 === RUN   Test_makeProbes_useExec
#33 70.20 --- PASS: Test_makeProbes_useExec (0.01s)
#33 70.20 === RUN   Test_makeProbes_useHTTPProbe
#33 70.20 --- PASS: Test_makeProbes_useHTTPProbe (0.00s)
#33 70.20 === RUN   Test_makeProbes_useCustomHTTPProbe
#33 70.20 --- PASS: Test_makeProbes_useCustomHTTPProbe (0.00s)
#33 70.20 === RUN   Test_makeProbes_useCustomDurationHTTPProbe
#33 70.20 --- PASS: Test_makeProbes_useCustomDurationHTTPProbe (0.00s)
#33 70.20 === RUN   Test_ParseProfileNames
#33 70.20 === RUN   Test_ParseProfileNames/empty_annotations_returns_nil_list
#33 70.20 === RUN   Test_ParseProfileNames/if_annotation_is_missing,_returns_nil_list
#33 70.20 === RUN   Test_ParseProfileNames/parses_profile_csv_string
#33 70.20 --- PASS: Test_ParseProfileNames (0.00s)
#33 70.20     --- PASS: Test_ParseProfileNames/empty_annotations_returns_nil_list (0.00s)
#33 70.20     --- PASS: Test_ParseProfileNames/if_annotation_is_missing,_returns_nil_list (0.00s)
#33 70.20     --- PASS: Test_ParseProfileNames/parses_profile_csv_string (0.00s)
#33 70.20 === RUN   Test_ProfilesToRemove
#33 70.20 === RUN   Test_ProfilesToRemove/empty_annotations_returns_nil_list
#33 70.20 === RUN   Test_ProfilesToRemove/requested_non-empty_and_existing_is_empty,_returns_nil_list
#33 70.20 === RUN   Test_ProfilesToRemove/missing_profile_annotation_on_request_and_existing,_returns_nil_list
#33 70.20 === RUN   Test_ProfilesToRemove/matching_profile_annotation_on_request_and_existing,_returns_nil_list
#33 70.20 === RUN   Test_ProfilesToRemove/overlapping_annotations_on_request_and_existing,_returns_extras_from_existing
#33 70.20 === RUN   Test_ProfilesToRemove/empty_annotation_on_request_and_non-empty_existing,_returns_all_existing_names
#33 70.20 --- PASS: Test_ProfilesToRemove (0.00s)
#33 70.20     --- PASS: Test_ProfilesToRemove/empty_annotations_returns_nil_list (0.00s)
#33 70.20     --- PASS: Test_ProfilesToRemove/requested_non-empty_and_existing_is_empty,_returns_nil_list (0.00s)
#33 70.20     --- PASS: Test_ProfilesToRemove/missing_profile_annotation_on_request_and_existing,_returns_nil_list (0.00s)
#33 70.20     --- PASS: Test_ProfilesToRemove/matching_profile_annotation_on_request_and_existing,_returns_nil_list (0.00s)
#33 70.20     --- PASS: Test_ProfilesToRemove/overlapping_annotations_on_request_and_existing,_returns_extras_from_existing (0.00s)
#33 70.20     --- PASS: Test_ProfilesToRemove/empty_annotation_on_request_and_non-empty_existing,_returns_all_existing_names (0.00s)
#33 70.20 === RUN   Test_TolerationsProfile_Apply
#33 70.20 --- PASS: Test_TolerationsProfile_Apply (0.00s)
#33 70.20 === RUN   Test_RunTimeClassProfile_Apply
#33 70.20 --- PASS: Test_RunTimeClassProfile_Apply (0.00s)
#33 70.20 === RUN   Test_RunTimeClassProfile_Remove
#33 70.20 === RUN   Test_RunTimeClassProfile_Remove/remove_matching_runtime_class_
#33 70.20 === RUN   Test_RunTimeClassProfile_Remove/leaves_runtime_class_that_does_not_match
#33 70.20 --- PASS: Test_RunTimeClassProfile_Remove (0.00s)
#33 70.20     --- PASS: Test_RunTimeClassProfile_Remove/remove_matching_runtime_class_ (0.00s)
#33 70.20     --- PASS: Test_RunTimeClassProfile_Remove/leaves_runtime_class_that_does_not_match (0.00s)
#33 70.20 === RUN   Test_TolerationsProfile_Remove
#33 70.20 --- PASS: Test_TolerationsProfile_Remove (0.00s)
#33 70.20 === RUN   Test_AffinityProfile_Apply
#33 70.20 --- PASS: Test_AffinityProfile_Apply (0.00s)
#33 70.20 === RUN   Test_AffinityProfile_Remove
#33 70.20 === RUN   Test_AffinityProfile_Remove/removes_matching_affinity_definition
#33 70.20 === RUN   Test_AffinityProfile_Remove/does_not_remove_non-matching_affinity_definition
#33 70.20 --- PASS: Test_AffinityProfile_Remove (0.00s)
#33 70.20     --- PASS: Test_AffinityProfile_Remove/removes_matching_affinity_definition (0.00s)
#33 70.20     --- PASS: Test_AffinityProfile_Remove/does_not_remove_non-matching_affinity_definition (0.00s)
#33 70.20 === RUN   Test_PodSecurityProfile_Apply
#33 70.20 --- PASS: Test_PodSecurityProfile_Apply (0.00s)
#33 70.20 === RUN   Test_PodSecurityProfile_Remove
#33 70.20 --- PASS: Test_PodSecurityProfile_Remove (0.00s)
#33 70.20 === RUN   Test_ConfigMapProfileParsing
#33 70.20 === RUN   Test_ConfigMapProfileParsing/unknown_profile_returns_error
#33 70.20 === RUN   Test_ConfigMapProfileParsing/yaml_profile_parsed_correctly
#33 70.20 === RUN   Test_ConfigMapProfileParsing/yaml_parsing_errors_are_returned
#33 70.20 --- PASS: Test_ConfigMapProfileParsing (0.04s)
#33 70.20     --- PASS: Test_ConfigMapProfileParsing/unknown_profile_returns_error (0.01s)
#33 70.20     --- PASS: Test_ConfigMapProfileParsing/yaml_profile_parsed_correctly (0.03s)
#33 70.20     --- PASS: Test_ConfigMapProfileParsing/yaml_parsing_errors_are_returned (0.00s)
#33 70.20 === RUN   Test_FunctionLookup
#33 70.20 === RUN   Test_FunctionLookup/function_without_namespace_uses_default_namespace
#33 70.20 === RUN   Test_FunctionLookup/function_with_namespace_uses_the_given_namespace
#33 70.20 === RUN   Test_FunctionLookup/url_parse_errors_are_returned
#33 70.20 --- PASS: Test_FunctionLookup (0.01s)
#33 70.20     --- PASS: Test_FunctionLookup/function_without_namespace_uses_default_namespace (0.00s)
#33 70.20     --- PASS: Test_FunctionLookup/function_with_namespace_uses_the_given_namespace (0.00s)
#33 70.20     --- PASS: Test_FunctionLookup/url_parse_errors_are_returned (0.00s)
#33 70.20 === RUN   Test_FunctionFactory_ConfigureSecrets
#33 70.20 --- PASS: Test_FunctionFactory_ConfigureSecrets (0.00s)
#33 70.20 === RUN   Test_configureReadOnlyRootFilesystem_Disabled_To_Disabled
#33 70.20 --- PASS: Test_configureReadOnlyRootFilesystem_Disabled_To_Disabled (0.00s)
#33 70.20 === RUN   Test_configureReadOnlyRootFilesystem_Disabled_To_Enabled
#33 70.20 --- PASS: Test_configureReadOnlyRootFilesystem_Disabled_To_Enabled (0.00s)
#33 70.20 === RUN   Test_configureReadOnlyRootFilesystem_Enabled_To_Disabled
#33 70.20 --- PASS: Test_configureReadOnlyRootFilesystem_Enabled_To_Disabled (0.00s)
#33 70.20 === RUN   Test_configureReadOnlyRootFilesystem_Enabled_To_Enabled
#33 70.20 --- PASS: Test_configureReadOnlyRootFilesystem_Enabled_To_Enabled (0.00s)
#33 70.20 PASS
#33 70.20 ok  	github.com/openfaas/faas-netes/pkg/k8s	0.514s
#33 70.20 === RUN   Test_makeApplyHandler
#33 70.20 I0115 21:46:04.075042    8752 apply.go:33] Deployment request for: nodeinfo
#33 70.20 I0115 21:46:04.092139    8752 apply.go:33] Deployment request for: nodeinfo
#33 70.20 I0115 21:46:04.092563    8752 apply.go:57] Updating nodeinfo
#33 70.20 --- PASS: Test_makeApplyHandler (0.06s)
#33 70.20 PASS
#33 70.20 ok  	github.com/openfaas/faas-netes/pkg/server	0.760s
#33 70.20 ?   	github.com/openfaas/faas-netes/pkg/signals	[no test files]
#33 70.20 ?   	github.com/openfaas/faas-netes/version	[no test files]
#33 70.20 FAIL
------
Dockerfile:21
--------------------
  19 |     RUN license-check -path /go/src/github.com/openfaas/faas-netes/ --verbose=false "Alex Ellis" "OpenFaaS Author(s)"
  20 |     RUN gofmt -l -d $(find . -type f -name '*.go' -not -path "./vendor/*")
  21 | >>> RUN CGO_ENABLED=${CGO_ENABLED} GOOS=${TARGETOS} GOARCH=${TARGETARCH} go test -v ./...
  22 |     
  23 |     RUN VERSION=$(git describe --all --exact-match `git rev-parse HEAD` | grep tags | sed 's/tags\///') \
--------------------
error: failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c CGO_ENABLED=${CGO_ENABLED} GOOS=${TARGETOS} GOARCH=${TARGETARCH} go test -v ./...]: exit code: 1
make: *** [Makefile:25: build-buildx-all] Error 1
```

**Comment by LucasRoesler** [comment]
For reference this is the bug I opened upstream in buildx https://github.com/docker/buildx/issues/509

**Comment by LucasRoesler** [comment]
Per this comment https://github.com/docker/buildx/issues/509#issuecomment-774239175 and conversations with @alexellis on slack, we should disable the tests when we are cross-compiling. I have added a small wrapper script to handle this logic . I ran the cross build locally and you can see it only runs the tests once. The second two test lines are only 0.1s long, indicating that they skip as expected 

image

**Comment by LucasRoesler** [comment]
If you wanted to go even further we could add a `skip tests` build arg and fully skip during the multi arch builds, save at least a minute there

**Comment by LucasRoesler** [comment]
bump @alexellis

**Comment by LucasRoesler** [comment]
@alexellis bump again

**Comment by LucasRoesler** [comment]
@alexellis i would like to either merge this or close it

File: Dockerfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ ARG TARGETPLATFORM
 ARG BUILDPLATFORM
 ARG TARGETOS
 ARG TARGETARCH
+ARG SKIP_TESTS
```
Comment (alexellis) [inline Dockerfile:9]: Can Go test skip the test that wasn't working whilst cross-compiling?

**Pull request closed without merging**